### DISCUSSION TITLE
Textare field not rendered on Index Page

### DISCUSSION THREAD

**Discussion by SteffenAuer** [laravel/nova-issues#4588] [Q&A] [answered]
- Laravel Version: 8.83.16
- Nova Version: 4.12.6
- PHP Version: 8.0.19
- Browser type and version: Firefox (**102.0.1**) & Safari (**15.5**)
- Reproduction Repository: _private_

### Description:

With the following code

```php
 Textarea::make('Name')
  ->alwaysShow()
  ->showOnIndex()
```

the textarea field only gets rendered on the detail & form pages, but the index page shows nothing.

<img width="88" alt="image" src="https://user-images.githubusercontent.com/60623116/179281902-8a04f4dc-990a-4558-a456-9b168f2baf7d.png">

When I inspect the element in the browser, the following is shown:
```html
﻿﻿<index-textarea-field class="text-left" field="[object Object]" resource="[object Object]" resource-name="achievement-levels"></index-textarea-field>
```

**Comment by henryavila** [comment]
@SteffenAuer 
TextArea is designed for big text that make no sense to display on Index page.

The bug is to display the column, when nothing should be displayed.

For some specific situation, I use this tooltip package, might be what you need
[https://github.com/elevenbytes/nova-tooltip-field](https://github.com/elevenbytes/nova-tooltip-field)
![image](https://user-images.githubusercontent.com/8429941/179285918-84a66713-7d25-4d27-a5ba-9659ecf1cdd3.png)

**Comment by davidhemphill** [comment, marked as answer]
It is not intended behavior for the Textarea to be seen on the index view. In the latest version of Nova, we'll throw an exception when calling this helper to inform the user instead of failing silently.

**Reply by halabudenco** [reply]
How to force display textarea?

**Reply by davidhemphill** [reply]
Nova doesn't support showing `Textarea` fields on the index views. You're free to use a `Text` field (set to only show on index) if you need to do this for some reason.

**Reply by halabudenco** [reply]
I did this and it helped 

```
 Text::make(__('Комментарий'),'comment')->displayUsing(function ($value) {
         return mb_substr($value,0,30,'UTF-8').'...';
})->onlyOnIndex(),

Textarea::make(__('Комментарий'),'comment')->hideFromIndex(),
```


Thanks

**Reply by Lemmings19** [reply]
Modified version that splits up the description to not grow too wide (limited to a 50 character width in this example) and also trim it only show the first 250 characters:
```php
TextArea::make(__('Description'), 'description')
    ->alwaysShow(),

Text::make(__('Description'))
    ->displayUsing(function ($value) {
        return nl2br(e(wordwrap(\Str::limit($value, 250), 50, "\n", true)));
    })
    ->asHtml()
    ->onlyOnIndex(),
```
